conn_linux.go

```diff
@@ -339,7 +342,7 @@ type sysSocket struct {
 // to a single thread.
 func newSysSocket(config *Config) (*sysSocket, error) {
 	// Determine network namespaces using the threadNetNS function.
-	g, err := newLockedNetNSGoroutine(config.NetNS, threadNetNS)
+	g, err := newLockedNetNSGoroutine(config.NetNS, threadNetNS, config.DisableNSLockThread)
```

Comment:
Double negatives get a bit confusing later, so how about `!config.DisableNSLockThread` here so the called functions can use the positive form instead?

Reply:
yes, I can see that. Pushing changes.